drivers/timers/wdt_notifier.c

```diff
@@ -0,0 +1,91 @@
+/****************************************************************************
+ * drivers/timers/wdt_notifier.c
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/arch.h>
+#include <nuttx/notifier.h>
+
+#include <sys/types.h>
+
+#include <nuttx/wdt_notifier.h>
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static ATOMIC_NOTIFIER_HEAD(g_wdt_notifier_list);
```

Comment:
merge into watchdog.c

Reply:
I did this by following panic_notifier.c. It might be better to put `g_wdt_notifier_list` in this file